service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -3,23 +3,49 @@
  */
 package com.lightbend.lagom.javadsl.server;
 
-import com.lightbend.lagom.internal.javadsl.*;
+import com.google.inject.Binder;
+import com.lightbend.lagom.internal.javadsl.BinderAccessor;
+import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServices;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServicesProvider;
 import com.lightbend.lagom.internal.javadsl.server.ServiceInfoProvider;
-import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl;
-
-import com.lightbend.lagom.javadsl.server.status.MetricsService;
-import com.google.inject.Binder;
 import com.lightbend.lagom.javadsl.api.ServiceInfo;
 import com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport;
+import com.lightbend.lagom.javadsl.server.status.MetricsService;
+
+import java.util.Arrays;
+import java.util.Optional;
 
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+    /**
+     * @deprecated use {@link ServiceGuiceSupport#bindServices(String, ServiceBinding[])} instead.
+     */
     default void bindServices(ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.empty(), serviceBindings);
+    }
+
+    /**
+     * binds the provided services and builds the {@link ServiceInfo} for this service.
+     *
+     * @param serviceName
+     * @param serviceBindings
+     */
+    default void bindServices(String serviceName, ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.of(serviceName), serviceBindings);
+    }
```

Comment:
I think then that we'll have to delay this change until Lagom 2.0.  I don't think duplicating the service name is acceptable. Runtime won't be fine since `bindServices(ServiceBinding, ServiceBinding...)` compiles to `bindServices(ServiceBinding, ServiceBinding[])`, which is a change in method signature from the current one which is `bindServices(ServiceBinding[])`, and so breaks binary compatibility.

I agree that enforcing at least one is good and should be done, but we can't break binary compatibility on a minor release.